src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -301,6 +301,12 @@ public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRe
                                     // Files tab is only supported in SQL Server, but files exists for all servers and used in detach database, cannot depend on files property to check the supportability
                                     ((DatabaseInfo)databaseViewInfo.ObjectInfo).IsFilesTabSupported = true;
                                     ((DatabaseInfo)databaseViewInfo.ObjectInfo).Filegroups = GetFileGroups(smoDatabase, databaseViewInfo);
+                                    try
+                                    {
+                                        // cannot retrieve ServerFilestreamAccessLevel property for localdb instance
```

Comment:
Do you have a link to docs that can help explain this? And did you chat with @shueybubbles about whether this is possibly something that could be fixed in SMO instead?

Reply:
filestreamlevel comes from the registry

[https://github.com/microsoft/sqlmanagementobjects/blob/d6f3e68fe6fae09d22355a160baa525d8032fd64/src/Microsoft/SqlServer/Management/SqlEnum/xml/inc_server.xml#L54C2-L54C2](https://github.com/microsoft/sqlmanagementobjects/blob/d6f3e68fe6fae09d22355a160baa525d8032fd64/src/Microsoft/SqlServer/Management/SqlEnum/xml/inc_server.xml#L54C2-L54C2)

is the comment referring specifically to the [localdb](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/sql-server-express-localdb?view=sql-server-ver16) product? If so, it's probably a bad sign that the docs for localdb expressly call it out as "can't be managed remotely via SQL Server Management Studio."
I suspect it is missing lots of functionality expected by SMO.
Are you trying to support localdb with ADS?

Reply:
@erinstellato-ms I've not heard of us explicitly targeting localdb with ADS - is that something we want to work to try and support?

Reply:
We don't need to support localdb unless @shueybubbles knows something I don't!

Reply:
OH! I thought we support localDB too, when i tested LocalDB for database properties, noticed this property cannot be retrieved, so added try/catch block to bypass the issue.

Reply:
I'm still confused to take an action on this one, please help me if I'm missing anything.

Before the change, I can open db properties on localhost localdb instance. with this change it is failing to load the properties and I thought it would be great to just bypass this parameter and not to affect the previous behavior. If we cannot use the logic for this scenario, then we will lose the working behavior.

So need your suggestions if I can leave the changes or remove the try/catch block and let ADS take care of failing the dialog to open. :)

Reply:
@ssreerama I'm not sure if this answers your question, but localdb is a version of SQLExpress with a wrapper that has a Windows service (so it's dependent on a Windows services to activate).  Visual Studio integrates with it in a couple ways, and it can be used with SSDT offline development.  ADS is cross platform.  If we are interested in supporting a lightweight local database option in ADS, completing our SQLLite data connector (used for testing) would be valuable to users, but I suspect (?) that's out of scope here.  However (yes, I'm going to complicate things), I've seen requests for supporting SQLLite in the NPS verbatims.  Do we want to go down that road?

@Charles-Gagnon  and @shueybubbles I'm going to tag you here for engineering help because I'm about out of my depth on how to advise Sai.

Also: I don't know if the work has been done by the appropriate team (which is not us) to ship localdb 2022.

Reply:
Hm, for a change this I guess we might as well fix it since it seems straightforward enough. But in general we should probably be careful not to spend too much time specifically trying to fix localdb issues until we decide to officially support it.